pkg/query-service/app/integrations/messagingQueues/kafka/sql.go

```diff
@@ -0,0 +1,171 @@
+package kafka
+
+import (
+	"fmt"
+)
+
+func generateConsumerSQL(start, end int64, topic, partition, consumerGroup, queueType string) string {
+	query := fmt.Sprintf(`
+WITH 
+-- Sub query for p99 calculation
+p99_query AS (
+    SELECT
+        serviceName,
+        quantile(0.99)(durationNano) / 1000000 as p99
+    FROM signoz_traces.signoz_index_v2
+    WHERE
+        timestamp >= '%d'
+      AND timestamp <= '%d'
+      AND kind = 5
+      AND msgSystem = '%s' 
+      AND stringTagMap['messaging.destination.name'] = '%s'
+      AND stringTagMap['messaging.destination.partition.id'] = '%s'
+	  AND stringTagMap['messaging.kafka.consumer.group'] = '%s'
+    GROUP BY serviceName
+),
+
+-- Sub query for RPS calculation
+rps_query AS (
+    SELECT
+        serviceName,
+        count(*) / ((%d - %d) / 1000000000) AS rps  -- Convert nanoseconds to seconds
+    FROM signoz_traces.signoz_index_v2
+    WHERE
+        timestamp >= '%d'
+        AND timestamp <= '%d'
+        AND kind = 5
+		AND msgSystem = '%s' 
+        AND stringTagMap['messaging.destination.name'] = '%s'
+        AND stringTagMap['messaging.destination.partition.id'] = '%s'
+		AND stringTagMap['messaging.kafka.consumer.group'] = '%s'
+    GROUP BY serviceName
+),
+
+-- Sub query for error rate calculation
+error_rate_query AS (
+    SELECT
+        serviceName,
+        count(*) / ((%d - %d) / 1000000000) AS error_rate  -- Convert nanoseconds to seconds
+    FROM signoz_traces.signoz_index_v2
+    WHERE
+        timestamp >= '%d'
+        AND timestamp <= '%d'
+        AND statusCode = 2
+        AND kind = 5
+        AND msgSystem = '%s' 
+        AND stringTagMap['messaging.destination.name'] = '%s'
+        AND stringTagMap['messaging.destination.partition.id'] = '%s'
+		AND stringTagMap['messaging.kafka.consumer.group'] = '%s'
+    GROUP BY serviceName
+),
+
+-- Sub query for average message size calculation
+avg_msg_size_query AS (
+    SELECT
+        serviceName,
+        avg(numberTagMap['messaging.message.body.size']) AS avg_msg_size
+    FROM signoz_traces.signoz_index_v2
+    WHERE
+        timestamp >= '%d'
+        AND timestamp <= '%d'
+        AND kind = 5
+        AND msgSystem = '%s' 
+        AND stringTagMap['messaging.destination.name'] = '%s'
+        AND stringTagMap['messaging.destination.partition.id'] = '%s'
+		AND stringTagMap['messaging.kafka.consumer.group'] = '%s'
```

Comment:
Will there ever be a case where `messaging.destination.*` exist but `messaging.message.body.size` doesn't?

Reply:
The Idea is to support users who setup following the guidelines, and in the guideline, we are making sure that the users install equal or higher version of the agents, if they do so, the fields are available by default.

Reply:
There could be an issue with average value calculations, specifically related to the `messaging.message.body.size` data.

Currently, if `messaging.message.body.size` does not exist in our numberTagMap, it returns zero. This can lead to incorrect averages because our query includes all rows, even when `messaging.message.body.size` doesn't exist. To fix this, we could add a check using `has(numberTagMap, 'messaging.message.body.size')`. This would ensure we only use rows where this data actually exists. You already have filters for `messaging.destination.*` and `messaging.kafka.consumer.group`. If these filters indirectly guarantee that `messaging.message.body.size` always exists, we might not need the extra check.

To be safe, though, it might be best to always include the has(...) check for `messaging.message.body.size`. This way, we're sure we're only using valid data in our calculations.

Reply:
done